clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java

```diff
@@ -37,11 +37,22 @@ public RequestHeader(Struct struct, short headerVersion) {
     }
 
     public RequestHeader(ApiKeys requestApiKey, short requestVersion, String clientId, int correlationId) {
-        this(new RequestHeaderData().
-                setRequestApiKey(requestApiKey.id).
-                setRequestApiVersion(requestVersion).
-                setClientId(clientId).
-                setCorrelationId(correlationId),
+        this(requestApiKey, requestVersion, clientId, correlationId, null, null);
+    }
+
+    public RequestHeader(ApiKeys requestApiKey,
```

Comment:
I think it would be good to have a constructor that didn't have the two forwarding fields, to avoid changing a large number of tests.

Reply:
We do have a constructor which doesn't contain the two fields above.